### PR TITLE
[LLM/LLD] Deprecated coins banner

### PR DESCRIPTION
**Pull request by lvndry**
### ❓ Context

- **JIRA or GitHub link**: [https://ledgerhq.atlassian.net/browse/LIVE-11925](https://ledgerhq.atlassian.net/browse/LIVE-11925)

### ✅ Checklist

- [x] `npx changeset` was attached.
- [x] **Covered by automatic tests.** 
- [x] **Impact of the changes:** 
  - LLD: when "will_be_deprecated" coin config is on, a banner should be displayed
  - LLM: when "will_be_deprecated" coin config is on, a banner should be displayed 

### 📝 Description

We add a banner in the account page to inform users a coin is about to be deprecated

Screenshot 2024-04-17 at 13 15 27

Screenshot 2024-04-17 at 13 21 26

Screenshot 2024-04-17 at 13 20 38

### ❓ Context

- **JIRA or GitHub link**: 

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

File: apps/ledger-live-desktop/src/renderer/components/TopBanner.tsx [hunk 1]
```diff
@@ -93,7 +113,14 @@ const TopBanner = ({ id, testId, content, status = "", dismissable = false, bann
         </IconContainer>
       )}
       {message}
-      <RightContainer>{right}</RightContainer>
+      {right && <RightContainer>{right}</RightContainer>}
+      {link && (
+        <LinkContainer>
+          <Link href={link.href} style={{ color: "black" }} alwaysUnderline>
```
Comment (hzheng-ledger) [inline apps/ledger-live-desktop/src/renderer/components/TopBanner.tsx:119]: just a question, have you tested the UI of both the light mode and dark mode?

File: apps/ledger-live-desktop/src/renderer/screens/account/index.tsx [hunk 2]
```diff
@@ -95,6 +106,7 @@ const AccountPage = ({
   const bgColor = useTheme().colors.palette.background.paper;
   const [shouldFilterTokenOpsZeroAmount] = useFilterTokenOperationsZeroAmount();
   const hiddenNftCollections = useSelector(hiddenNftCollectionsSelector);
+  const [currencyConfig, setCurrencyConfig] = useState<CurrencyConfig & Record<string, unknown>>();
```
Comment (hzheng-ledger) [inline apps/ledger-live-desktop/src/renderer/screens/account/index.tsx:109]: just a question, can we just 
```
const [currencyConfig, setCurrencyConfig] = useState<CurrencyConfig>();
```
Is the "Record<string, unknown>" useful?

File: apps/ledger-live-desktop/src/renderer/screens/account/index.tsx [hunk 3]
```diff
@@ -109,12 +121,28 @@ const AccountPage = ({
...
   );
 
+  const currency = getAccountCurrency(account);
+
+  useEffect(() => {
+    try {
+      if (isCryptoCurrency(currency)) {
+        const currencyConfig = getCurrencyConfiguration(currency);
+        setCurrencyConfig(currencyConfig);
+        console.log({ currencyConfig });
```
Comment (hzheng-ledger) [inline apps/ledger-live-desktop/src/renderer/screens/account/index.tsx:131]: "console.log" is not necessary

File: apps/ledger-live-desktop/src/renderer/screens/account/index.tsx [hunk 4]
```diff
@@ -109,12 +121,28 @@ const AccountPage = ({
     [hiddenNftCollections, shouldFilterTokenOpsZeroAmount],
   );
 
+  const currency = getAccountCurrency(account);
+
+  useEffect(() => {
+    try {
```
Comment (hzheng-ledger) [inline apps/ledger-live-desktop/src/renderer/screens/account/index.tsx:127]: I assume that we don't need to use hooks(useEffect and the useState for currencyConfig) for currencyConfig.
we can just use a variable "currencyConfig"
```
const currencyConfig = getCurrencyConfiguration(currency);
```

File: apps/ledger-live-mobile/src/screens/Account/index.tsx [hunk 5]
```diff
@@ -112,9 +107,22 @@ const AccountScreenInner = ({
 
   const { secondaryActions } = useAccountActions({ account, parentAccount });
```
Comment (hzheng-ledger) [inline apps/ledger-live-mobile/src/screens/Account/index.tsx:109]: same comment as I did for LLD

File: apps/ledger-live-mobile/src/screens/Account/index.tsx [hunk 6]
```diff
@@ -149,7 +157,11 @@ const AccountScreenInner = ({
   return (
     <ReactNavigationPerformanceView screenName={ScreenName.Account} interactive>
       <SafeAreaView isFlex>
-        {analytics}
+        <TrackScreen
```
Comment (hzheng-ledger) [inline apps/ledger-live-mobile/src/screens/Account/index.tsx:160]: I think it change is not necessary, right?

File: apps/ledger-live-mobile/src/screens/WalletCentricAsset/index.tsx [hunk 7]
```diff
@@ -98,6 +104,17 @@ const AssetScreen = ({ route }: NavigationProps) => {
     }
   }, [currency, navigation]);
 
+  useEffect(() => {
```
Comment (hzheng-ledger) [inline apps/ledger-live-mobile/src/screens/WalletCentricAsset/index.tsx:107]: same comment as I did for LLD

File: libs/coin-modules/coin-bitcoin/src/bridge/js.ts [hunk 8]
```diff
@@ -88,7 +89,14 @@ function buildAccountBridge(signerContext: SignerContext, perfLogger: PerfLogger
...
   startSpan: StartSpan;
 };
-export function createBridges(signerContext: SignerContext, perfLogger: PerfLogger) {
+
+export function createBridges(
+  signerContext: SignerContext,
+  perfLogger: PerfLogger,
+  coinConfig: CoinConfig,
+) {
+  setCoinConfig(coinConfig);
```
Comment (hzheng-ledger) [inline libs/coin-modules/coin-bitcoin/src/bridge/js.ts:98]: we need to add "ledger-live/libs/ledger-live-common/src/families/bitcoin/config.ts"
for the default config value for the deprecated coins. Right?

**Review by hzheng-ledger — CHANGES_REQUESTED** [review]
Have a look at my comments, plz.
Thanks

File: libs/coin-modules/coin-bitcoin/src/bridge/js.ts (hunk 8 above)
Comment (lvndry) [reply libs/coin-modules/coin-bitcoin/src/bridge/js.ts:98]: Yeah the deprecated coins already have default values in config.ts

Comment (lvndry) [reply libs/coin-modules/coin-bitcoin/src/bridge/js.ts:98]: https://github.com/LedgerHQ/ledger-live/blob/deprecated-coins-banner/libs/ledger-live-common/src/families/bitcoin/config.ts#L212

We let them as "active" for now and we'll change accordingly on Firebase when the time is right

**Review by hzheng-ledger — DISMISSED** [review]
It looks good. Thanks

**Review by hzheng-ledger — APPROVED** [review]
looks good

**Pull request merged**